### DISCUSSION TITLE
Trying to use FastDDS with QtCreator and MinGW [17538]

### DISCUSSION THREAD

**Discussion by cerberus77** [eProsima/Fast-DDS#3314] [Unofficial support]
I'm trying to use FastDDS with my Qt project. I'm using MinGW compiler. The project compiles, however I got following errors during linking.

`:-1: error: debug/main.o:C:\projects\CASCUpdaterTool\src\build-SoftwareUpdaterApp-Desktop_Qt_6_4_2_MinGW_64_bit-Debug/../libs/fastrtps-2.9.1/include/fastdds/rtps/common/Time_t.h:458: undefined reference to `eprosima::fastrtps::rtps::Time_t::Time_t(int, unsigned int)'
:-1: error: debug/main.o: in function `__static_initialization_and_destruction_0':
C:\projects\CASCUpdaterTool\src\libs\fastrtps-2.9.1\include\fastdds\rtps\common\Time_t.h:459: error: undefined reference to `eprosima::fastrtps::rtps::Time_t::Time_t(int, unsigned int)'
:-1: error: C:\projects\CASCUpdaterTool\src\build-SoftwareUpdaterApp-Desktop_Qt_6_4_2_MinGW_64_bit-Debug/../libs/fastrtps-2.9.1/include/fastdds/rtps/common/Time_t.h:460: undefined reference to `eprosima::fastrtps::rtps::Time_t::Time_t(int, unsigned int)'
:-1: error: C:\projects\CASCUpdaterTool\src\build-SoftwareUpdaterApp-Desktop_Qt_6_4_2_MinGW_64_bit-Debug/../libs/fastrtps-2.9.1/include/fastdds/rtps/common/Time_t.h:679: undefined reference to `eprosima::fastrtps::Time_t::Time_t(int, unsigned int)'
:-1: error: C:\projects\CASCUpdaterTool\src\build-SoftwareUpdaterApp-Desktop_Qt_6_4_2_MinGW_64_bit-Debug/../libs/fastrtps-2.9.1/include/fastdds/rtps/common/Time_t.h:681: undefined reference to `eprosima::fastrtps::Time_t::Time_t(int, unsigned int)'
:-1: error: C:\projects\CASCUpdaterTool\src\build-SoftwareUpdaterApp-Desktop_Qt_6_4_2_MinGW_64_bit-Debug/../libs/fastrtps-2.9.1/include/fastdds/rtps/common/Time_t.h:683: undefined reference to `eprosima::fastrtps::Time_t::Time_t(int, unsigned int)'
:-1: error: collect2.exe: error: ld returned 1 exit status
:-1: error: [Makefile.Debug:128: debug/SoftwareUpdaterApp.exe] Error 1`

Any advice how to fix these? It's always eprosima::fastrtps::rtps::Time_t::Time_t object which causes the linking problems.

**Comment by richiware** [comment]
Likely the symbols is not imported. In order to import Fast-DDS's DLL symbols use the preprocessor definition FASTRTPS_DYN_LINK.
